test/e2e/ssm/command.go

```diff
@@ -82,3 +88,34 @@ func RunCommand(ctx context.Context, client *ssm.Client, instanceId, command str
 
 	return commandOutput, nil
 }
+
+// there is no built in wait for instance to be registered with ssm
+// WaitForInstance uses DescribeInstanceInformation in a loop to wait for it be registered
+// see: https://github.com/aws/aws-cli/issues/4006
+func WaitForInstance(ctx context.Context, client *ssm.Client, instanceId string, logger logr.Logger) error {
+	waitCtx, cancel := context.WithTimeout(ctx, instanceRegisterTimeout)
+	defer cancel()
+
+	logger.Info(fmt.Sprintf("Waiting for instance %s to be registered with SSM", instanceId))
+	for {
+		output, err := client.DescribeInstanceInformation(waitCtx, &ssm.DescribeInstanceInformationInput{
+			Filters: []types.InstanceInformationStringFilter{
+				{
+					Key:    aws.String("InstanceIds"),
+					Values: []string{instanceId},
+				},
+			},
+		})
+		if err != nil {
+			return err
+		}
+		if len(output.InstanceInformationList) > 0 {
+			logger.Info(fmt.Sprintf("Instance %s is registered with SSM", instanceId))
```

Comment:
nit: with structured loggers it's usually not recommended to do string interpolation
```suggestion
			logger.Info(fmt.Sprintf("Instance is registered with SSM", "instanceID", instanceId))
```